### ISSUE TITLE
missing subtype rules when deferring types separate from classes

### ISSUE THREAD

**Issue by sigmundch** [dart-lang/sdk#43055]
Consider this program:

**a.dart**:
```dart
import 'b.dart' deferred as b;
import 'c.dart';

main() async {
  var f = () => b.C1();
  print(f is C2 Function());
  print(f is C3 Function());
  await b.loadLibrary();
}
```

**b.dart**:
```dart
import 'c.dart';
class C1 extends C2 implements C3 {}
```

**c.dart**:
```dart
class C2 {}
class C3 {}
```

The program should print `true` and `true`. Today it prints `false` and `false` due to the unsoundness without `--new-deferred-split`. 
```
> dart2js --disable-program-split # correctly prints true true
> dart2js --no-defer-class-types # incorrectly prints false false
> dart2js --defer-class-types # incorrectly prints false false
> dart2js --new-deferred-split --no-defer-class-types # correctly prints true true
```

It appears that the new algorithm to separate classes from types is also missing some subtyping information:
```
> dart2js --new-deferred-split --defer-class-types # incorrectly prints false false
```

Inspecting the output, it appears that the relation between `C1` and it's supertypes is emitted in the deferred fragment.

**Comment by sigmundch** [comment]
@fishythefish - quick question here. I believe we just need the subtyping rules to address these issues. Are there places where the rti implementation relies on the `inherit` calls to derive subtyping data? Or is it usually replicated in the subtyping tables?

**Comment by fishythefish** [comment]
No, all (interface) subtyping data should be reflected in the table. Sometimes `x is T` is lowered to a check for `x.$isT`, which could be affected by `inherit`, but a check between two types should be unaffected.

**Comment by sigmundch** [comment]
this has been fixed, and since 29c0795f48ff8c12aa0793c0fbfef201eb423aed the regression test is passing.
